resolver/resolver.cc

```diff
@@ -1522,6 +1893,13 @@ class ResolveTypeMembersWalk {
             resolveAttachedClass(ctx, job.klass, resolvedAttachedClasses);
         }
 
+        // Resolve simple field declarations. Required so that `type_alias` can refer to an enum value type
+        // (which is a static field).
```

Comment:
Do you mind saying that this is a stronger claim than we need? We only need that static fields used by `T::Enum` are resolved, right?

Also, I think that we technically could have implemented that logic precisely by asking whether the RHS of the static field assignment had an enum NameRef with `isTEnumName`.